pkg/controller/namespace/namespace_controller.go

```diff
@@ -132,7 +132,9 @@ func (nm *NamespaceController) enqueueNamespace(obj interface{}) {
 		glog.Errorf("Couldn't get key for object %+v: %v", obj, err)
 		return
 	}
-	nm.queue.Add(key)
+	// delay processing namespace deletions long enough for the lifecycle admission
+	// plugins to also observe the deletion and stop new objects from being created
+	nm.queue.AddAfter(key, 5*time.Second)
```

Comment:
> On a different note, it seems we need a way to identify if admission has run

can you clarify what you mean by that? admission exists on every API server in HA scenarios, admission only runs on a deletion on the API server that handles the delete request, and anything persisted in etcd has run through admission already